tracer/src/Datadog.Trace/Iast/TaintedObjects.cs

```diff
@@ -9,19 +9,49 @@ namespace Datadog.Trace.Iast
 {
     internal class TaintedObjects
     {
+        private static readonly bool _largeNumericCache = false;
         private readonly ITaintedMap _map;
 
+        static TaintedObjects()
+        {
+            // From Net 8.0 onwards first 300 digit strings are cached instead of first 10
+            _largeNumericCache = object.ReferenceEquals(299.ToString(), 299.ToString());
+        }
+
         public TaintedObjects()
         {
             _map = new DefaultTaintedMap();
         }
 
-        public void TaintInputString(string stringToTaint, Source source)
+        public bool TaintInputString(string stringToTaint, Source source)
         {
-            if (!string.IsNullOrEmpty(stringToTaint))
+            if (!IsFiltered(stringToTaint))
```

Comment:
you are not longer checking if `stringToTaint` is null in `IsFiltered`

Reply:
String should not be null here (#nullable enable). I've checked all the use places